server/base/src/main/java/org/apache/accumulo/server/compaction/CompactionJobGenerator.java

```diff
@@ -205,9 +209,22 @@ private Collection<CompactionJob> planCompactions(CompactionServiceId serviceId,
         // remove any files that are in active compactions
         tablet.getExternalCompactions().values().stream().flatMap(ecm -> ecm.getJobFiles().stream())
             .forEach(tmpFiles::remove);
-        // remove any files that are selected
-        if (tablet.getSelectedFiles() != null) {
-          tmpFiles.keySet().removeAll(tablet.getSelectedFiles().getFiles());
+        // remove any files that are selected and the user compaction has completed
+        // at least 1 job, otherwise we can keep the files
+        var selectedFiles = tablet.getSelectedFiles();
+
+        long selectedExpirationDuration =
+            ConfigurationTypeHelper.getTimeInMillis(env.getConfiguration(tablet.getTableId())
+                .get(Property.TABLE_COMPACTION_SELECTION_EXPIRATION.getKey()));
+
+        // If jobs are completed, or selected time has not expired, the remove
+        // from the candidate list otherwise we can cancel the selection
+        if (selectedFiles != null) {
+          if (selectedFiles.getCompletedJobs() > 0 || (selectedFiles.getSelectedTime() != null
+              && (System.nanoTime() - selectedFiles.getSelectedTime().getNanos())
```

Comment:
Seems like nanoTime and steady time are being compared.  Also selectedExpirationDuration is in millils, so need to use millils when comparing.

```suggestion
              && steadyTime.minus(selectedFiles.getSelectedTime()).toMillis()
```

Reply:
Thanks for pointing that out, yeah this was a mistake, I noticed this too today as I was re-looking at this when my IDE was giving me a warning about the SteadyTime object that is a private member never being used. The intent was to use it here just like that and it got missed.